ipld/hamt/src/pointer.rs

```diff
@@ -37,25 +56,58 @@ impl<K: PartialEq, V: PartialEq, H> PartialEq for Pointer<K, V, H> {
     }
 }
 
+mod pointer_v0 {
+    use cid::Cid;
+    use serde::Serialize;
+
+    use crate::KeyValuePair;
+
+    use super::Pointer;
+
+    #[derive(Serialize)]
+    pub(super) enum PointerSer<'a, K, V> {
+        Vals(&'a [KeyValuePair<K, V>]),
+        Link(&'a Cid),
+    }
```

Comment:
I'm a bit confused at how this works. For v0, we want to serialize to a map of "0"/"1" to a cid/list of links. As far as I know, this here will serialize use "Vals" and "Link" as the keys?